chain-network/proto/node.proto

```diff
@@ -104,6 +104,14 @@ message PullBlocksToTipRequest {
   repeated bytes from = 1;
 }
 
+// Request message for method PullBlocks
+message PullBlocksRequest {
+  // The identifiers of a starting block.
+  bytes from = 1;
+  // The identifier of the end block
+  bytes to = 2;
+}
+
```

Comment:
This is confusingly different from `PullHeadersRequest` and `PullBlocksToTipRequest`: the `from` field in those requests is repeated, meaning to give a list of checkpoints.

Reply:
Will `from_id` work here?

Reply:
I'd like to keep all similar requests with homogeneous parameter fields.
That said, a non-repeated field in protobuf can be converted to repeated [in a backward compatible way][protobuf-updating] (the doc talks about optional fields, which means any field in proto3). So maybe we can keep this `from` bound as a non-repeating field for now. It will even work correctly (but wastefully) if, in a future update, a client sends a repeating `from` field with the most remote checkpoint as the last member, but the server replaces all repeating values with the last one it receives.

Another option is to just make the field repeating now and implement the method the same way as we do `PullHeaders`. We just won't use multiple `from` checkpoints in our node, but the protocol will look consistent.

[protobuf-updating]: https://developers.google.com/protocol-buffers/docs/proto#updating

Reply:
I am leaning towards the first option. From the practical point of view (request a range of blocks) having a repeated field does not make any sense. To make it less confusing we could just rename the method to `PullBlocksRange` 🤷